src/core/proto/route_table_mgr.cpp

```diff
@@ -152,113 +155,117 @@ void route_table_mgr::update_tbl(nl_data_t data_type)
     netlink_socket_mgr::update_tbl(data_type);
 }
 
-void route_table_mgr::parse_entry(struct nlmsghdr *nl_header)
+void route_table_mgr::parse_entry(struct nl_object *nl_obj)
 {
-    int len;
-    struct rtmsg *rt_msg;
-    struct rtattr *rt_attribute;
     route_val val;
 
-    // get route entry header
-    rt_msg = (struct rtmsg *)NLMSG_DATA(nl_header);
+    // Cast the generic nl_object to a specific route or rule object
+    struct rtnl_route *route = reinterpret_cast<struct rtnl_route *>(nl_obj);
 
-    if (rt_msg->rtm_family != AF_INET && rt_msg->rtm_family != AF_INET6) {
-        return;
-    }
+    val.set_family(rtnl_route_get_family(route));
+    val.set_protocol(rtnl_route_get_protocol(route));
+    val.set_scope(rtnl_route_get_scope(route));
+    val.set_type(rtnl_route_get_type(route));
+    val.set_table_id(rtnl_route_get_table(route));
 
-    val.set_family(rt_msg->rtm_family);
-    val.set_protocol(rt_msg->rtm_protocol);
-    val.set_scope(rt_msg->rtm_scope);
-    val.set_type(rt_msg->rtm_type);
-    val.set_table_id(rt_msg->rtm_table);
-    val.set_dst_pref_len(rt_msg->rtm_dst_len);
+    // Set destination mask and prefix length
+    struct nl_addr *dst = rtnl_route_get_dst(route);
+    if (dst) {
+        val.set_dst_pref_len(nl_addr_get_prefixlen(dst));
+    }
 
-    len = RTM_PAYLOAD(nl_header);
-    rt_attribute = (struct rtattr *)RTM_RTA(rt_msg);
+    parse_attr(route, val);
 
-    for (; RTA_OK(rt_attribute, len); rt_attribute = RTA_NEXT(rt_attribute, len)) {
-        parse_attr(rt_attribute, val);
-    }
     val.set_state(true);
 
     route_table_t &table = val.get_family() == AF_INET ? m_table_in4 : m_table_in6;
     table.push_back(val);
 }
 
-void route_table_mgr::parse_attr(struct rtattr *rt_attribute, route_val &val)
+void route_table_mgr::parse_attr(struct rtnl_route *route, route_val &val)
 {
-    char if_name[IFNAMSIZ];
+    struct nl_addr *addr;
 
-    switch (rt_attribute->rta_type) {
-    case RTA_DST:
-        val.set_dst_addr(ip_address((void *)RTA_DATA(rt_attribute), val.get_family()));
-        break;
-    // next hop address
-    case RTA_GATEWAY:
-        val.set_gw(ip_address((void *)RTA_DATA(rt_attribute), val.get_family()));
-        break;
-    // unique ID associated with the network interface
-    case RTA_OIF:
-        val.set_if_index(*(int *)RTA_DATA(rt_attribute));
-        if_indextoname(val.get_if_index(), if_name);
+    // Destination Address
+    addr = rtnl_route_get_dst(route);
+    if (addr) {
+        val.set_dst_addr(ip_address(nl_addr_get_binary_addr(addr), val.get_family()));
+    }
+
+    // Gateway Address (Next Hop)
+    struct rtnl_nexthop *nh = rtnl_route_nexthop_n(route, 0); // Assuming the first nexthop
```

Comment:
since libnl treats non-multipath nexthop as multipath internally, can we assume that that the multipath loop below handles all the cases and current code block is redundant?

Reply:
relying on this internal behavior may not be robust in the long term. 

The internal implementation details of libnl could change in future updates, which might break our code if we assume this behavior. 

To ensure the robustness and maintainability of our code, it is safer to explicitly handle non-multipath nexthop cases separately. This approach will protect us against any potential changes in libnl's internal handling of nexthops.